File: fractrangenerator.py
dimx = 30
dimy = 30

# generate primes to use as variables
primes = [2, 3]
def nextPrime():
    p = primes[-1]
    while True:
        p+=2
        if checkPrime(p):
            primes.append(p)
            return
def checkPrime(p):
    for q in primes:
        if p%q == 0:
            return False
    return True
while(len(primes) < 3*dimx*dimy+2):
    nextPrime()

variables = []
for i in range(3):
    variables.append([])
    for j in range(dimx):
        variables[i].append([])
        for k in range(dimy):
            variables[i][j].append(primes[i*dimx*dimy + j*dimy + k + 2])


program = ""
program += str(primes[0]) + "/" + str(primes[1]) + ", "

for i in range(dimx):
    for j in range(dimy):
        nominator = primes[1]*variables[2][i][j]
        for x in (1, 0, -1):
            xindex = i+x
            if xindex==-1 or xindex==dimx:
                continue
            for y in (1, 0, -1):
                yindex = j+y
                if yindex==-1 or yindex==dimy or (x==0 and y==0):
                    continue
                nominator *= variables[1][xindex][yindex]
            
        program += str(nominator) + "/" + str(variables[0][i][j] * primes[0]) + ", "

program += "1/"+str(primes[0]) + ", "

for i in range(dimx):
    for j in range(dimy):
        var0 = variables[0][i][j]
        var1 = variables[1][i][j]
        var2 = variables[2][i][j]
        
        program += "1/" + str(var1**7) + ", "
        program += "1/" + str(var1**6) + ", "
        program += "1/" + str(var1**4) + ", "
        program += str(var0) + "/" + str(var1**3) + ", "
        program += str(var0) + "/" + str(var1**2*var2) + ", "
        program += "1/" + str(var1) + ", "
        program += "1/" + str(var2) + ", "
        
        
print(program[:-2])

file = open("fractrancode.txt", "w+")
file.write(program[:-2])
file.close()









        

File: src/com/company/FractranGenerator.java
package com.company;

import java.io.FileNotFoundException;
import java.io.PrintWriter;
import java.io.UnsupportedEncodingException;
import java.math.BigInteger;
import java.util.ArrayList;
import java.util.Arrays;

import static jdk.nashorn.internal.objects.NativeString.substring;

public class FractranGenerator {
    static ArrayList<Long> primes = new ArrayList<>(Arrays.asList((long) 2, (long) 3));

    static void nextPrime()
    {
        long p = primes.get(primes.size() - 1);
        while (true)
        {
            p += 2;
            if(checkPrime(p))
            {
                primes.add(p);
                return;
            }
        }
    }

    static boolean checkPrime(long p)
    {
        for (long q : primes)
        {
            if(p%q == 0)
            {
                return false;
            }
        }
        return true;
    }

    static public void generateProgram(int xd, int yd) throws FileNotFoundException, UnsupportedEncodingException {
        while (primes.size() < 3*xd*yd + 2)
        {
            nextPrime();
        }

        BigInteger[][][] variables = new BigInteger[3][xd][yd];
        for(int i = 0; i < 3; i++)
        {
            for(int j = 0; j < xd; j++)
            {
                for (int k = 0; k < yd; k++)
                {
                    variables[i][j][k] = BigInteger.valueOf(primes.get(i*xd*yd + j*yd + k + 2));
                }
            }
        }

        String program = "";
        BigInteger nominator;
        program += primes.get(0).toString() + "/" + primes.get(1).toString() + ", ";
        for(int i =0; i < xd; i++)
        {
            for(int j =0; j < yd; j++)
            {
                nominator = variables[2][i][j].multiply(BigInteger.valueOf(primes.get(1)));
                for(int x = -1; x < 2; x++)
                {
                    int xindex = i + x;
                    if(xindex == -1 || xindex == xd)
                    {
                        continue;
                    }
                    for(int y = -1; y < 2; y++)
                    {
                        int yindex = j + y;
                        if(yindex == -1 || yindex == yd || (x == 0 && y == 0))
                        {
                            continue;
                        }
                        nominator = nominator.multiply(variables[1][xindex][yindex]);

                    }
                }

                program += nominator.toString() + "/" + variables[0][i][j].multiply(BigInteger.valueOf(primes.get(0))) + ", ";

            }

        }
        program += "1/" + primes.get(0).toString() + ", ";

        BigInteger var0;
        BigInteger var1;
        BigInteger var2;
        for(int i = 0; i < xd; i++)
        {
            for(int j = 0; j < yd; j++)
            {

                var0 = variables[0][i][j];
                var1 = variables[1][i][j];
                var2 = variables[2][i][j];

                program += "1/" + var1.pow(7).toString() + ", ";
                program += "1/" + var1.pow(6).toString() + ", ";
                program += "1/" + var1.pow(4).toString() + ", ";
                program += var0.toString() + "/" + var1.pow(3).toString() + ", ";
                program += var0.toString() + "/" + var1.pow(2).multiply(var2).toString() + ", ";
                program += "1/" + var1.toString() + ", ";
                program += "1/" + var2.toString() + ", ";
            }

        }
        PrintWriter writer = new PrintWriter("fractrancode.txt");
        writer.print(substring(program, 0, program.length() - 2));
        writer.close();


    }
}
